packages/actor-init-query/lib/ActorInitQueryBase.ts

```diff
@@ -126,5 +146,25 @@ export interface IActorInitQueryBaseArgs extends IActorInitArgs {
    *   "localizeBlankNodes": "@comunica/actor-query-operation:localizeBlankNodes"
    * }}
    */
-  contextKeyShortcuts: Record<string, string>;
+  contextKeyShortcuts: Record<string, string> | Partial<Record<keyof QueryContext, string>>;
+  /**
+   * An array of `contextKeyShortcuts` that are to be appended to the (default) `contextKeyShortcuts`
+   * (which are by default injected by component.js).
+   *
+   * The appending happens in the constructor call. Conflicting keys will cause an error.
+   * If you extend `ActorInitQueryBase` and want to add custom shortcuts, do so as follows:
```

Comment:
This is what we need indeed! But I would go a step further, and make it so that the TS code below is not needed anymore.
Using Components.js annotations magic, we should be able to define these extensions via JSDoc annotations, similar to [here](https://github.com/comunica/comunica/blob/master/packages/actor-init-query/lib/ActorInitQueryBase.ts#L97-L128).

Reply:
I'm not quite sure how to achieve this with the annotation magic referenced here (and couldn't find documentation).

Since the goal after all would be to extend the annotated default context (in the current case with `contextKeyShortcutsExtensions`), I am not sure how this would work for chained inherits (without overriding `contextKeyShortcutsExtensions` which would make the `Record<string, string>[]` array useless).

Am I getting that wrong or missing something?

Reply:
You're right, this isn't possible yet using Components.js annotations.
We'd have to extend the Components.js generator to make this array append logic possible.
Instead of doing that, it would be better to just extend Components.js to extend JSON objects.

But in any case, that's beyond the scope of this PR, so let's leave that for future work. So you can ignore my comment here :-)